### PR TITLE
FYLE-cqf2uk-1: Search, Sort and Filter in Reports page

### PR DESCRIPTION
**Pull request by Dimple16**
Screenshot 2021-10-01 at 8 26 02 PM
Screenshot 2021-10-01 at 8 26 25 PM
Screenshot 2021-10-01 at 8 26 56 PM
Screenshot 2021-10-01 at 8 27 18 PM
Screenshot 2021-10-01 at 8 30 32 PM
Screenshot 2021-10-01 at 8 30 47 PM
Screenshot 2021-10-01 at 8 30 58 PM

### CODE REVIEW

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 1]
```diff
@@ -464,4 +511,494 @@ export class MyReportsPage implements OnInit {
...
+        generatedFilters.sortParam = 'rp_purpose';
+        generatedFilters.sortDir = 'asc';
+      } else if (sortBy.value === 'nameZToA') {
+        generatedFilters.sortParam = 'rp_purpose';
+        generatedFilters.sortDir = 'desc';
+      }
+    }
+  }
+
+  convertFilters(selectedFilters: SelectedFilters<any>[]): Filters {
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:699]: you can extract convertFilters & convertSelectedSortFitlersToFilters to a service

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 2]
```diff
@@ -464,4 +511,494 @@ export class MyReportsPage implements OnInit {
...
+    this.convertSelectedSortFitlersToFilters(sortBy, generatedFilters);
+
+    return generatedFilters;
+  }
+
+  generateStateFilterPills(filterPills: FilterPill[], filter) {
+    filterPills.push({
+      label: 'State',
+      type: 'state',
+      value: filter.state
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:725]: inquiry reports are called sent back now. can you check that case once

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 3]
```diff
@@ -464,4 +511,494 @@ export class MyReportsPage implements OnInit {
...
+    }
+
+    const sortBy = selectedFilters.find((filter) => filter.name === 'Sort By');
+
+    this.convertSelectedSortFitlersToFilters(sortBy, generatedFilters);
+
+    return generatedFilters;
+  }
+
+  generateStateFilterPills(filterPills: FilterPill[], filter) {
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:721]: move this too into a service

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 4]
```diff
@@ -464,4 +511,494 @@ export class MyReportsPage implements OnInit {
...
+
+  async openFilters(activeFilterInitialName?: string) {
+    const filterPopover = await this.modalController.create({
+      component: FyFiltersComponent,
+      componentProps: {
+        filterOptions: [
+          {
+            name: 'State',
+            optionType: FilterOptionType.multiselect,
+            options: [
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:900]: get the options from a service

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 5]
```diff
@@ -464,4 +511,494 @@ export class MyReportsPage implements OnInit {
...
+
+  searchClick() {
+    this.headerState = HeaderState.simpleSearch;
+    const searchInput = this.simpleSearchInput.nativeElement as HTMLInputElement;
+    setTimeout(() => {
+      searchInput.focus();
+    }, 300);
+  }
+
+  convertRptDtSortToSelectedFilters(
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:571]: a bunch of these methods are pure functions and can be moved into a service

**Review by Julias0 — APPROVED** [review]

File: src/app/fyle/my-reports/my-reports.page.ts (hunk 5 above)
Comment (Dimple16) [reply src/app/fyle/my-reports/my-reports.page.ts:571]: refactoring filter methods in other pr

File: src/app/fyle/my-reports/my-reports.page.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/fyle/my-reports/my-reports.page.ts:725]: yes, this is sent back

**Pull request merged**